instrumentation/ruby_kafka/test/opentelemetry/instrumentation/ruby-kafka/patches/consumer_test.rb

```diff
@@ -0,0 +1,133 @@
+# frozen_string_literal: true
+
+# Copyright The OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+require 'test_helper'
+
+require_relative '../../../../../lib/opentelemetry/instrumentation/ruby_kafka'
+require_relative '../../../../../lib/opentelemetry/instrumentation/ruby_kafka/patches/consumer'
+
+describe OpenTelemetry::Instrumentation::RubyKafka::Patches::Consumer do
+  let(:instrumentation) { OpenTelemetry::Instrumentation::RubyKafka::Instrumentation.instance }
+  let(:exporter) { EXPORTER }
+  let(:spans) { exporter.finished_spans }
+
+  let(:host) { ENV.fetch('TEST_KAFKA_HOST') { '127.0.0.1' } }
+  let(:port) { (ENV.fetch('TEST_KAFKA_PORT') { 29_092 }) }
+
+  let(:kafka) { Kafka.new(["#{host}:#{port}"], client_id: 'opentelemetry-kafka-test') }
+  let(:topic) { "topic-#{SecureRandom.uuid}" }
+  let(:producer) { kafka.producer }
+  let(:consumer) { kafka.consumer(group_id: SecureRandom.uuid, fetcher_max_queue_size: 1) }
+
+  before do
+    kafka.create_topic(topic)
+    consumer.subscribe(topic)
+
+    # Clear spans
+    exporter.reset
+
+    instrumentation.install
+  end
+
+  after do
+    # Force re-install of instrumentation
+    instrumentation.instance_variable_set(:@installed, false)
+
+    # Clean up
+    producer.shutdown
+    consumer.stop
+    kafka.close
+  end
+
+  describe '#each_message' do
+    it 'traces each_meassage call' do
+      kafka.deliver_message('hello', topic: topic)
+      kafka.deliver_message('hello2', topic: topic)
+
+      begin
+        counter = 0
+        consumer.each_message do |_msg|
+          counter += 1
+          raise 'oops' if counter >= 2
+        end
+      rescue StandardError # rubocop:disable Lint/HandleExceptions
+      end
+
+      process_spans = spans.select { |s| s.name == "#{topic} process" }
+
+      # First pair for send and process spans
+      first_process_span = process_spans[0]
+      _(first_process_span.name).must_equal("#{topic} process")
+      _(first_process_span.kind).must_equal(:consumer)
+      _(first_process_span.attributes['messaging.destination']).must_equal(topic)
+      _(first_process_span.attributes['messaging.kafka.partition']).must_equal(0)
+
+      first_process_span_link = first_process_span.links[0]
+      linked_span_context = OpenTelemetry::Trace.current_span(first_process_span_link.span_context).context
```

Comment:
Oh, I see - this is treating the `link.span_context` as a `Context` instead of a `SpanContext`. This (and other examples below) should look like:
```suggestion
      linked_span_context = first_process_span_link.span_context
```

Reply:
Kinda feels like we should be more strict about type-checking the `span_context` argument in the `Link` initializer.

Reply:
Yeah I agree, it's a bit too easy to do the wrong thing if you're not entirely familiar with that part of the code.